tests/test_pure_resource_path.py

```diff
@@ -0,0 +1,186 @@
+from sublime_lib import ResourcePath
+
+from unittest import TestCase
+
+
+class TestPureResourcePath(TestCase):
+
+    def test_eq(self):
+        self.assertEqual(
+            ResourcePath("Packages/Foo/bar.py"),
+            ResourcePath("Packages/Foo/bar.py")
+        )
+
+    def test_eq_slash(self):
+        self.assertEqual(
+            ResourcePath("Packages/Foo/bar.py"),
+            ResourcePath("Packages/Foo/bar.py///")
+        )
+
+    def test_str(self):
+        self.assertEqual(
+            str(ResourcePath("Packages/Foo/bar.py")),
+            "Packages/Foo/bar.py"
+        )
+
+    def test_repr(self):
+        self.assertEqual(
+            repr(ResourcePath("Packages/Foo/bar.py")),
+            "ResourcePath('Packages/Foo/bar.py')"
+        )
+
+    def test_parts(self):
+        path = ResourcePath("Packages/Foo/bar.py")
+        self.assertEqual(path.parts, ("Packages", "Foo", "bar.py"))
+
+    def test_parts_empty(self):
+        path = ResourcePath("")
+        self.assertEqual(path.parts, ())
+
+    def test_parent(self):
+        self.assertEqual(
+            ResourcePath("Packages/Foo/bar.py").parent,
+            ResourcePath("Packages/Foo")
+        )
+
+    def test_top_parent(self):
+        self.assertEqual(
+            ResourcePath("Packages").parent,
+            ResourcePath("Packages")
```

Comment:
Why? Shouldn't this give `ResourcePath("")`?
The root is the data dir after all.

```
In [28]: PurePath("abc").parent
Out[28]: PurePosixPath('.')
```

Reply:
I guess this depends on our definition. If we think of resource paths more like Windows paths with the drive names as roots, it would make sense to return the first part on `.root` and to not go beyond single-part paths.

I am fine with either, both have pros and cons. However, we should document what we think of when talking about the "root" and .parent and .root should have aligning behavior.